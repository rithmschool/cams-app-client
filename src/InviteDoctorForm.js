import React, {Component} from 'react';
import {BASE_URL, userID, config} from './helpers.js';
import axios from 'axios';

class InviteDoctorForm extends Component {
    constructor(props) {
        super(props)
        this.state = {
            email: '',
            response: ""
        }
    }

    inviteDoctor(config, thisArg) {
        axios.post(`${BASE_URL}/api/users/invite`, {
            email: thisArg.state.email
        }, config() )
        .then(response => {
            this.setState({
                response: response.data
            });
        })
        .catch(response => {
            this.setState({
                response: response.response.data
            });
        })
        this.setState({
            response: "One moment..."
        });
    }

    handleChange(e) {
        this.setState({
            [e.target.name]: e.target.value
        });
    }

    handleSubmit(e) {
        e.preventDefault();
        if (this.state.email) {
            this.inviteDoctor(config, this)
            this.setState({
                email: ''
            })
        }
    }

    render () {
    let responseText = <p>{this.state.response}</p>
    return (
      <div>
        <div className="banner-text">
          <h1 className="banner-bold">Invite</h1>
        </div>
        <div className="content">
          <h1>Invite New Doctor</h1>
          <form onSubmit={this.handleSubmit.bind(this)}>
            <input
              type="email"
              name="email"
              placeholder="email"
              value={this.state.email}
              onChange={this.handleChange.bind(this)}
            />
            <button className="button button-hover" type="submit" value="Submit">Send Invite</button>
          </form>
          {responseText}
      </div>
    </div>
    )
  }
}

export default InviteDoctorForm;