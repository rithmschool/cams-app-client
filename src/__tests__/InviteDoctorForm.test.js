import InviteDoctorForm from "../InviteDoctorForm";
import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";

test("matches snapshot of InviteDoctorForm", () => {
  const inviteDoctorForm = renderer.create(<InviteDoctorForm />).toJSON();
  expect(inviteDoctorForm).toMatchSnapshot();
});

test("should pass a selected value to the onSubmit handler", () => {
  const spy = jest.spyOn(InviteDoctorForm.prototype, "handleSubmit");
  const wrapper = shallow(<InviteDoctorForm />);
  wrapper.find("form").simulate("submit", { preventDefault() {} });
  expect(spy).toHaveBeenCalled();
});

it("responds to a change", () => {
  const spy = jest.spyOn(InviteDoctorForm.prototype, "handleChange");
  const wrapper = shallow(<InviteDoctorForm />);
  wrapper.find('input[type="email"]').simulate("change", {
    target: {
      name: "email",
      value: "s"
    }
  });
  expect(spy).toHaveBeenCalled();
});
