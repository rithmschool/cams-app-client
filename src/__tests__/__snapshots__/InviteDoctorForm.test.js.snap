// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of InviteDoctorForm 1`] = `
<div>
  <div
    className="banner-text"
  >
    <h1
      className="banner-bold"
    >
      Invite
    </h1>
  </div>
  <div
    className="content"
  >
    <h1>
      Invite New Doctor
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
      <button
        className="button button-hover"
        type="submit"
        value="Submit"
      >
        Send Invite
      </button>
    </form>
    <p>
      
    </p>
  </div>
</div>
`;
